### PR TITLE
Init ui::TabControl

### PR DESCRIPTION
**Pull request by geron-cn**
Init ui::TabControl

### CODE REVIEW

**Comment by geron-cn** [comment]
[ciskip]

**Comment by geron-cn** [comment]
@zilongshanren  @liamcindy  could you please review this pr,
@chengstory  win8.1 configure lack only

**Comment by geron-cn** [comment]
[cirebuild]

**Comment by xiaofeng11** [comment]
[ci rebuild]

File: cocos/ui/UITabControl.h [hunk 1]
```diff
@@ -0,0 +1,326 @@
...
+        //Color4B      _backgroundColor;
+        enum FontType
+        {
+            SYSTEM,
+            TTF,
+            BMFONT
+        };
+        FontType  _fontType;
+        
+        /* CustomCommand _backgroundCmd;
```
Comment (zilongshanren) [inline cocos/ui/UITabControl.h]: Please remove these unneeded comments.

File: cocos/ui/UITabControl.h [hunk 2]
```diff
@@ -0,0 +1,326 @@
...
+         * @param headerCell, the header Button, will be a protected child in TabControl
+         * @param the container, will be a protected child in TabControl
+         */
+        void      insertTab(int index, TabHeader* headerCell, Layout* container);
+        
+        // the tab count
+        size_t    getTabCount() const;
+        
+        // get current selected tab's index
+        inline int  getCurrentTabIndex() { return _currItemIndex; }
```
Comment (zilongshanren) [inline cocos/ui/UITabControl.h]: please add const to all the getters, thanks.

**Comment by geron-cn** [comment]
@zilongshanren  test case added
@chengstory   test case file need win8.1 project configure

**Comment by geron-cn** [comment]
[ci rebuild]

File: cocos/ui/UITabControl.h [hunk 3]
```diff
@@ -0,0 +1,388 @@
...
+    /**
+     * the header button in TabControl
+     */
+    class CC_GUI_DLL TabHeader : public AbstractCheckButton
+    {
+        friend class TabControl;
+        
+    public:
+        
+        enum EventType
```
Comment (zilongshanren) [inline cocos/ui/UITabControl.h:53]: This should be enum class? we prefer strong typed enum over the old enum..

File: cocos/ui/UITabControl.h [hunk 4]
```diff
@@ -0,0 +1,388 @@
...
+        
+    private:
+        Label*       _tabLabelRender;
+        float        _tabLabelFontSize;
+        TabControl*  _tabView;
+        
+        typedef std::function<void(int tabindex, TabHeader::EventType)> ccTabCallback;
+        ccTabCallback  _tabSelectedEvent;
+        
+        enum FontType
```
Comment (zilongshanren) [inline cocos/ui/UITabControl.h:181]: here, please also change enum to `enum class`.

**Pull request closed without merging**